src/settingsdialog.h

```diff
@@ -0,0 +1,43 @@
+#ifndef SETTINGSDIALOG_H
+#define SETTINGSDIALOG_H
+
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QAbstractButton>
+#include "qwebstyleedit.h"
+#include "mainwindow.h"
+#include "parsers/perf/perfparser.h"
+#include "resultspage.h"
+
+namespace Ui {
+class SettingsDialog;
+}
+
+class SettingsDialog : public QDialog
+{
+    Q_OBJECT
+
+public:
+    MainWindow* mainWindow;
```

Comment:
make private, = nullptr

Reply:
remove, the dialog should be fed from the mainwindow with the parameters and then we query back for the new values when the dialog got accepted (e..g via a signal)

don't set the params directly on the mainwindow from the settings dialog